src/cpr_data_access/utils.py

```diff
@@ -86,3 +86,29 @@ def dig(obj: Union[list, dict], *fields: Any, default: Any = None) -> Any:
         elif not obj:
             return default
     return obj
+
+
+def unflatten_json(data: dict) -> dict:
+    """
+    Unflatten a dictionary with keys that are dot-separated strings.
+
+    I.e. metadata.data respresents {"metadata": {"data": {}}}
+    """
+    unflattened = {}
+    for key, value in data.items():
+        parts = key.split(".")
+        current = unflattened
+        for part in parts[:-1]:
+            current = current.setdefault(part, {})
+        current[parts[-1]] = value
+    return unflattened
+
+
+def remove_if_all_nested_vals_none(data: dict, key: str) -> dict:
```

Comment:
naming suggestion: `remove_key_if_all_nested_vals_none`. I think this then helps you understand what the second arg does when using this method if it's not passed with the `key` keyword